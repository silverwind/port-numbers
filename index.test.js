import {getService, getPort} from "./index.js";

test("getService", () => {
  expect(getService(12345).name).toEqual("italk");
  expect(getService(3306).name).toEqual("mysql");
  expect(getService(53, "udp").name).toEqual("domain");
  expect(getService(65536)).toEqual(null);
  expect(getService(68).description).toEqual("Bootstrap Protocol Client");
  expect(getService(70).description).toEqual("Gopher");
  expect(getService(80).name).toEqual("http");
  expect(getService(80, "tcp").name).toEqual("http");
  expect(getService(9).description).toEqual("Discard");
});

test("getPort", () => {
  expect(getPort("doesnotexist")).toEqual(null);
  expect(getPort("http").port).toEqual(80);
  expect(getPort("https").port).toEqual(443);
  expect(getPort("italk").port).toEqual(12345);
  expect(getPort("mysql").port).toEqual(3306);
  expect(getPort("ntp").port).toEqual(123);
  expect(getPort("snmp").description).toEqual("SNMP");
  expect(getPort("ssh").description).toEqual("The Secure Shell (SSH) Protocol");
  expect(getPort("tftp").description).toEqual("Trivial File Transfer");
});
