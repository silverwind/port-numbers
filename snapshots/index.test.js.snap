// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ports 1`] = `
[
  "",
  "Reserved",
]
`;

exports[`ports 2`] = `
[
  "mysql",
  "MySQL",
]
`;

exports[`ports 3`] = `
[
  "domain",
  "Domain Name Server",
]
`;

exports[`ports 4`] = `undefined`;

exports[`ports 5`] = `
[
  "www-http",
  "World Wide Web HTTP",
]
`;
